lib/galaxy_social.py

```diff
@@ -108,18 +112,35 @@ def parse_markdown_file(self, file_path):
             else {}
         )
 
+        mentions_invalid = metadata["mentions"].keys() - metadata["media"]
+        if mentions_invalid:
+            errors += f"- Mentions for `{', '.join(mentions_invalid)}` social medias are not in medias list of metadata.\n"
+
+        hashtags_invalid = metadata["hashtags"].keys() - metadata["media"]
+        if hashtags_invalid:
+            errors += f"- Hashtags for `{', '.join(hashtags_invalid)}` social medias are not in medias list of metadata.\n"
+
         image_pattern = re.compile(r"!\[(.*?)\]\((.*?)\)")
         images = image_pattern.findall(text)
         plain_content = re.sub(image_pattern, "", text).strip()
 
+        for image in images:
+            try:
+                if requests.get(image[1]).status_code != 200:
+                    errors += f"- Image `{image[1]}` not found.\n"
+            except:
+                errors += f"- Invalid Image url `{image[1]}`.\n"
+
```

Comment:
:+1: on the rest, but downloading images as part of linting seems weird and is too expensive, imo.

Since images will be visible in the preview (on github at least, it's also rather easy for the user to spot. Better just fail the entire real posting if an image cannot be downloaded when it's needed.

Reply:
Fixed with the new commit.